app/controllers/participate/login.js

```diff
@@ -15,31 +15,31 @@ export default Ember.Controller.extend({
   authenticating: false,
   actions: {
     authenticate(attrs) {
-      return this.get('session')
-        .authenticate('authenticator:jam-jwt', attrs)
-        .then(() => {
-          var surveyController = Ember.getOwner(this).lookup('controller:participate');
-          surveyController.set('studyId', attrs.password);
-          surveyController.set('participantId', attrs.username);
-          this.transitionToRoute('participate.survey.consent');
-        })
-        .catch((e) => {
-          if (e.status === 404) {
-            this.send('toggleInvalidAuth');
+      if (!this.get('locale')) {
+        this.toggleProperty('invalidLocale');
+        this.set('authenticating', false);
+        this.get('raven').captureMessage('Locale not selected');
+      } else {
+        return this.get('session')
+          .authenticate('authenticator:jam-jwt', attrs)
+          .then(() => {
+            var surveyController = Ember.getOwner(this).lookup('controller:participate');
+            surveyController.set('studyId', attrs.password);
+            surveyController.set('participantId', attrs.username);
+            this.transitionToRoute('participate.survey.consent');
+          })
+          .catch((e) => {
             this.set('authenticating', false);
-          } else if (e.name === 'TransitionAborted') {
-            this.send('toggleInvalidLocale');
-            this.get('raven').captureMessage('Locale not selected', {
-              status: e.status
-            });
-          }
-        });
+            if (e.status === 404 || e.status === 401) {
```

Comment:
If the error code is a `404` or `401`, the user should get a message saying "Invalid study ID or participant ID", otherwise they should get some generic error message. The else block is here: https://github.com/CenterForOpenScience/isp/pull/103/files#diff-7998c4c951082a36d95902b6ae2275b6R35